pkg/instancegroups/instancegroups.go

```diff
@@ -561,3 +561,21 @@ func (c *RollingUpdateCluster) deleteNode(ctx context.Context, node *corev1.Node
 
 	return nil
 }
+
+func (c *RollingUpdateCluster) RollSingleInstance(ctx context.Context, cloudMember *cloudinstances.CloudInstanceGroupMember, detach bool) error {
+	if detach {
+		if cloudMember.CloudInstanceGroup.InstanceGroup.IsMaster() {
+			return fmt.Errorf("cannot detatch master instances. Run with --detatch=false")
```

Comment:
I think it would be better to just ignore `detach` in this case. Having to reissue the command with a difficult to type flag is not the best user experience.